app/scripts/controllers/main.js

```diff
@@ -31,12 +26,25 @@ angular.module('scoreTrackerApp')
           $scope.scoreForm.$setPristine();
         }
         $scope.newScore = { name: "", value: undefined };
-
-        updateList();
-        $scope.updateSummary();
-
+        $scope.updateList();
         focusForm();
       }
     };
 
+    $scope.$watch('scores', function() {
```

Comment:
It is better for a controller to have as little logic as possible (only for view support). The rational is that services are easier to test and reuse. You probably will never reuse a controller within a directive.

Also this deep watch is not very efficient. It will evaluate at least twice on every `$digest` involving a change.

The score service knows about scores in the system and about actions that can be run against it (add, remove, update) and it can calculate min, max and average (caches at this point) based on those actions. The view is then just observing these changes. The controller shouldn't have to coordinate actions against the score service